src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/DotnetCliHelper.cs

```diff
@@ -69,15 +69,15 @@ private async Task<string> GetDotnetSdkFolderFromDotnetExecutableAsync(string pr
         return dotnetSdkFolderPath;
     }
 
-    public Process Run(string[] arguments, string? workingDirectory, bool shouldLocalizeOutput, bool redirectStandardInput = false)
+    public Process Run(string[] arguments, string? workingDirectory, bool shouldLocalizeOutput)
     {
         _logger.LogDebug($"Running dotnet CLI command at {_dotnetExecutablePath.Value} in directory {workingDirectory} with arguments {arguments}");
 
         var startInfo = new ProcessStartInfo(_dotnetExecutablePath.Value)
         {
             CreateNoWindow = true,
             UseShellExecute = false,
-            RedirectStandardInput = redirectStandardInput,
+            RedirectStandardInput = true,
```

Comment:
We probably then need to have an explicit close of the standard input stream; otherwise something that were to try to read is now going to wait (forever) for that to complete.

Reply:
The component which would block, is the child dotnet process which is calling `Console.ReadLine()` or similar, right?

Are you asking to bring back a flag like `bool leaveStandardInputOpen` or some such, so that this component can close it if needed?